src/django_subatomic/db.py

```diff
@@ -81,13 +114,17 @@ def transaction_if_not_already(*, using: str | None = None) -> Generator[None]:
     # If the innermost atomic block is from a test case, we should create a SAVEPOINT here.
     # This allows for a rollback when an exception propagates out of this block, and so
     # better simulates a production transaction behaviour in tests.
-    savepoint = _innermost_atomic_block_wraps_testcase(using=using)
-
-    with (
-        _execute_on_commit_callbacks_in_tests(using),
-        django_transaction.atomic(using=using, savepoint=savepoint),
-    ):
+    if in_transaction(using=using):
+        # Already in a transaction; don't open another.
         yield
+    else:
+        # If the innermost atomic block comes from the testcase, create a SAVEPOINT.
+        savepoint = _innermost_atomic_block_wraps_testcase(using=using)
+        with _error_if_pending_testcase_on_commit_callbacks(using=using):
+            with _execute_on_commit_callbacks_in_tests(using), django_transaction.atomic(
+                using=using, savepoint=savepoint
+            ):
+                yield
```

Comment:
I don't think we want the `in_transaction` here:

```suggestion
    savepoint = _innermost_atomic_block_wraps_testcase(using=using)

    with (
        _error_if_pending_testcase_on_commit_callbacks(using=using),
        _execute_on_commit_callbacks_in_tests(using),
        django_transaction.atomic(using=using, savepoint=savepoint),
    ):
        yield
```